### PR TITLE
Moving RecurringTasks.cs to separate library  

### PR DESCRIPTION
**Pull request by isaac73**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Moving RecurringTasks.cs to separate library  
...

**Checklist:**

- [x] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: RecurringTasksAgent/RecurringTasks.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 using Microsoft.Azure.IoTSolutions.IotHubManager.Services;
 using Microsoft.Azure.IoTSolutions.IotHubManager.Services.Diagnostics;
 
-namespace Microsoft.Azure.IoTSolutions.IotHubManager.DevicePropertiesAgent
+namespace Microsoft.Azure.IoTSolutions.IotHubManager.RecurringTasksAgent
```
Comment (dluc) [inline RecurringTasksAgent/RecurringTasks.cs]: for consistency with other projects, could you rename the class to `Agent` and the interface to `IRecurringTasksAgent` please?

File: RecurringTasksAgent/Agent.cs [hunk 2]
```diff
@@ -36,63 +36,63 @@ public class RecurringTasks : IRecurringTasks
...
                     this.deviceProperties.TryRecreateListAsync().Wait(CACHE_TIMEOUT_SECS * 1000);
-                    this.log.Info("Cache created", () => { });
+                    this.log.Info("DeviceProperties Cache created", () => { });
                     return;
                 }
-                catch (Exception e)
+                catch (Exception)
                 {
-                    this.log.Warn("Cache creation failed, will retry in few seconds", () => new { CACHE_INIT_RETRY_SECS, e });
+                    this.log.Debug("DeviceProperties Cache creation failed, will retry in few seconds", () => new { CACHE_INIT_RETRY_SECS});
```
Comment (dluc) [inline RecurringTasksAgent/Agent.cs]: why this change?

Comment (isaac73) [reply RecurringTasksAgent/Agent.cs]: This is too verbose when storage adapter is not available. There is already enough info in logs from here - > https://github.com/Azure/iothub-manager-java/blob/master/app/com/microsoft/azure/iotsolutions/iothubmanager/RecurringTasksAgent/RecurringTasks.java
